app/controllers/concerns/scopable.rb

```diff
@@ -0,0 +1,23 @@
+module Scopable
+  module Controller
+    extend ActiveSupport::Concern
+
+    # Add has_scope for each model attribute
+    included do
+      controller_name.classify.constantize.attribute_names.each do |a|
```

Comment:
J'ajouterai une méthode pour plus de clarté ici : `controller_attributes.each do ...`

Reply:
vu que c'est dans un block `included do` il y a un problème de visibilité des méthodes, du coup ça ne marche pas 'out of the box'.

En call j'avais donné mon avis qu'on pouvait laisser comme ça du coup :/

je laisse @Samuelfaure confirmer c'est lui qui sait :)

Reply:
Non ça ne marche pas en effet x) tenter un workaround rajouterais la complexité qu'on cherche à enlever je pense

Reply:
Je parle juste de refactor ces quelques lignes, un truc du genre ça devrait marcher je pense : 

```ruby 
included do 
  controller_attributes.each { |attr| add_scope_for_attribute(attr) } 
end 

private 

def controller_attributes
  controller_name.classify.constantize.attribute_names
end

def add_scope_for_attribute(attr)
  has_scope(attr.to_sym, ->(value) { where(Hash[attr, value]) }, only: :index)
end
```

Reply:
Pas mal ! Ceci fonctionne :

```ruby
    included do
      controller_attributes.each { |attr| add_has_scope_for_attribute(attr) }
    end

    module ClassMethods
      private

      def controller_attributes
        controller_name.classify.constantize.attribute_names
      end

      def add_has_scope_for_attribute(attr)
        has_scope(attr.to_sym, ->(value) { where(Hash[attr, value]) }, only: :index)
      end
    end
```

Reply:
Parfait.

Reply:
ok, sans le `ClassMethods`... ça me fonctionnait pas
bien joué
 :+1: